test/object-store/sync/flx_sync.cpp

```diff
@@ -2173,30 +2153,30 @@ TEST_CASE("flx: asymmetric sync", "[sync][flx][app]") {
 
         harness->do_with_new_user([&](std::shared_ptr<SyncUser> user) {
             SyncTestFile config(user, schema, SyncConfig::FLXSyncEnabled{});
-            std::condition_variable cv;
-            std::mutex wait_mutex;
-            bool wait_flag(false);
-            std::error_code ec;
-            config.sync_config->error_handler = [&](std::shared_ptr<SyncSession>, SyncError error) {
-                std::unique_lock<std::mutex> lock(wait_mutex);
-                wait_flag = true;
-                ec = error.error_code;
-                cv.notify_one();
+            auto [error_promise, error_future] = util::make_promise_future<SyncError>();
+            auto shared_promise = std::make_shared<decltype(error_promise)>(std::move(error_promise));
```

Comment:
Is this needed or could a `CopyablePromiseHolder` be used?

Reply:
CopyablePromiseHolder can probably be used. I just took inspiration from other tests.